api/core/workflow/nodes/base/node.py

```diff
@@ -56,6 +57,40 @@ class Node:
     node_type: ClassVar["NodeType"]
     execution_type: NodeExecutionType = NodeExecutionType.EXECUTABLE
 
+    # Global registry populated via __init_subclass__
+    _registry: ClassVar[dict["NodeType", dict[str, type["Node"]]]] = {}
+
+    def __init_subclass__(cls, **kwargs):  # type: ignore[override]
+        super().__init_subclass__(**kwargs)
+        # Skip base class itself
+        if cls is Node:
+            return
+        # Only register concrete subclasses that define node_type and version()
+        node_type = getattr(cls, "node_type", None)
+        version_fn = getattr(cls, "version", None)
+        if node_type is None or not callable(version_fn):
+            return
+        try:
+            version = cls.version()
+        except Exception:
+            # Ignore subclasses that cannot provide a version at import time
+            return
+        bucket = Node._registry.setdefault(node_type, {})
+        bucket[version] = cls  # type: ignore[index]
+        # Maintain a "latest" pointer preferring numeric versions; fallback to lexicographic
+        version_keys = [v for v in bucket if v != "latest"]
+        numeric_pairs: list[tuple[str, float]] = []
+        for v in version_keys:
+            try:
+                numeric_pairs.append((v, float(v)))
```

Comment:
All versions are integers now. So it's ok to use `int` to convert version string to version number.